src/DarkNet/controllers/ServerGenerator.ts

```diff
@@ -261,47 +236,51 @@ export const getGuessNumberServer = (difficulty: number, x: number, y: number):
     password: password,
     staticPasswordHint: `The password is a number between 0 and ${maxNumber}`,
     difficulty,
-    x,
-    y,
+    depth: depth,
+    leftOffset: leftOffset,
   });
 };
 
-export const getLargeDictionaryServer = (difficulty: number, x: number, y: number): DarknetServer => {
+export const getLargeDictionaryServer = (difficulty: number, depth: number, leftOffset: number): DarknetServer => {
   return getDictionaryAttackServer(
     difficulty,
-    x,
-    y,
+    depth,
+    leftOffset,
     commonPasswordDictionary,
     ["It's a common password"],
     Minigames.CommonPasswordDictionary,
   );
 };
 
-export const getEuCountryDictionaryServer = (difficulty: number, x: number, y: number): DarknetServer => {
+export const getEuCountryDictionaryServer = (difficulty: number, depth: number, leftOffset: number): DarknetServer => {
   return getDictionaryAttackServer(
     difficulty,
-    x,
-    y,
+    depth,
+    leftOffset,
     EUCountries,
     ["My favorite EU country"],
     Minigames.EUCountryDictionary,
   );
 };
 
-export const getYesn_tServer = (difficulty: number, x: number, y: number): DarknetServer => {
+export const getYesn_tServer = (difficulty: number, depth: number, leftOffset: number): DarknetServer => {
```

Comment:
Looking at all of these.... there is a *lot* of boilerplate. What about something like this (needs better name):

```suggestion
const ServerBuilderBuilder(params_func) => {
  return (difficulty: number, depth: number, leftOffset: number): DarknetServer => {
    return DnetServerBuilder({
      icon: getRandomIcon(),
      difficulty,
      depth,
      leftOffset,
      ...params_func(difficulty),
    });
  }
}

export const getYesn_tServer = ServerBuilderBuilder(difficulty => ({
  modelId: Minigames.Yesn_t,
  password: getPassword(3 + difficulty / 2, true, difficulty > 8, false, false),
  staticPasswordHint: "you are one who's'nt authorized",
}));
```

Reply:
I've done something similar to this (converted the server builders to just make the relevant few config options and removed the server boilerplate from them)